src/components/progress-circle/progress_circle.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  Component,
+  ViewEncapsulation,
+  ElementRef,
+  Attribute,
+  Input,
+  HostBinding
+} from 'angular2/core';
+import {isPresent, CONST} from 'angular2/src/facade/lang';
+import {OneOf} from '../../core/annotations/one-of';
+
+
+// TODO(josephperrott): Benchpress tests.
+
+/** Display modes of Progress Circle */
+@CONST()
+class ProgressMode {
+  @CONST() static DETERMINATE = 'determinate';
+  @CONST() static INDETERMINATE = 'indeterminate';
+}
+
+
+/**
+ * <md-progress-circle> component.
+ */
+@Component({
+  selector: 'md-progress-circle',
+  host: {
+    'role': 'progressbar',
+    'aria-valuemin': '0',
+    'aria-valuemax': '100',
+  },
+  templateUrl: './components/progress-circle/progress_circle.html',
+  styleUrls: ['./components/progress-circle/progress-circle.css'],
+})
+export class MdProgressCircle {
+  /**
+   * Value of the progress circle.
+   *
+   * Input:number, defaults to 0.
+   * value_ is bound to the host as the attribute aria-valuenow.
+   */
+  @Input('value')
+  value_: number = 0;
+  @HostBinding('attr.aria-valuenow')
+  get _value() {
+    return this.value_;
+  }
+
+  /**
+   * Mode of the progress circle
+   *
+   * Input must be one of the values from ProgressMode, defaults to 'determinate'.
+   * mode is bound to the host as the attribute host.
+   */
+  @Input()
+  @OneOf([ProgressMode.DETERMINATE, ProgressMode.INDETERMINATE])
+  mode: string;
+  @HostBinding('attr.mode')
+  get _mode() {
+    return this.mode;
+  }
+
+
+  /**
+   * Gets the current stroke dash offset to represent the progress circle.
+   *
+   * The stroke dash offset specifies the distance between dashes in the circle's stroke.
+   * Setting the offset to a percentage of the total circumference of the circle, fills this
+   * percentage of the overall circumference of the circle.
+   */
+  strokeDashOffset() {
+    /**
+     * To determine how far the offset should be, we multiple the current percentage by the
+     * total circumference.
+     *
+     * The total circumference is calculated based on the radius we use, 45.
+     *    PI * 2 * 45
+     */
```

Comment:
This one should be an inline-style comment (since its not describing a method or property)